### ISSUE TITLE
[a11y Support]: Secure Messaging Alignment with Design System Guidelines

### ISSUE THREAD

**Issue by sara-amanda** [department-of-veterans-affairs/va.gov-team#62820]
### Point of contact

Sara Smith & Angela Fowler

### Severity level

3, Moderate. Should be fixed in 1-3 sprints post-launch.

### Details

> ## Platform Issue Identified
> 
> Design components or patterns don't align with Design System guidelines.
> 
> #### Issue Details
> 
> When printing messages in a conversation, the radio button group for selecting one message or all messages doesn't have a legend to describe the radio options. When testing with NVDA and tabbing to the radio buttons, it's announced as an "undefined grouping." It looks like the va-radio component doesn't have a label property and has an aria-label of "undefined."
> 
> **VA.gov Experience Standard**
> [Category Number 04, Issue Number 07](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/VA.gov-experience-standards.1683980311.html)
> 
> #### Other References
> 
> WCAG SC 3.2.4 AA

## Reproduction Steps
- [ ] Able to reproduce the issue
- [x] Unable to reproduce the issue

## a11y Update
We partnered with Angela Fowler on this ticket to test the screen reader issue. @AngelaFowler82 I went ahead and documented what Juan relayed for you in the ticket, as it relates to [JIRA MHV-4606]( https://jira.devops.va.gov/browse/MHV-46066), that we do not have access to. 

### Original Inquiry
**[Shared via Slack](https://dsva.slack.com/archives/C05HSMK3NCA/p1689777750313679) with @laurwill by Juan 7/19/2023:** 

> [The ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/59484) states that the screen reader reads "undefined grouping" and that the component doesn't have a label property, and the suggestion was to add a subtitle (line over the radial buttons). How would that be done without creating a redundancy, as in having the word print listed twice?

### Unable to Duplicate the Issue

**c/o Juan 7/28/2023**
> I was able to talk to [Angela Fowler](https://dsva.slack.com/team/U01MTTZB3PZ) about the issue and **we were not able to replicate the issue with NVDA screen screen reader** (as it was reported on the Jira ticket) nor other screen reader that she tested. 

> **It may no longer be an issue as the modal nor the radio buttons in staging are not producing errors.**

### Proposed solution or next steps

> ## Platform Recommendation
> 
> Note that the design system's [guidance on radio buttons](https://design.va.gov/components/form/radio-button#accessibility-considerations) includes an instruction to have a legend, and the component in Storybook has the label property marked as required (which populates the legend).
> 
> I'm guessing the label was left off because it would be redundant with the modal heading, right? I know there are some teams (thinking of the VA Forms team) that have used headings also function as the legend, although that's in the context of the forms library and not in a modal component. I would encourage you to explore something like that, or otherwise to work on some kind of content adjustment so that there can be both the heading and a valid legend.

### Platform Resources
> **VA.gov Experience Standard**
> [Category Number 04, Issue Number 07](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/VA.gov-experience-standards.1683980311.html)
> 
> #### Other References
> 
> WCAG SC 3.2.4 AA

### Shared Resources

**(@sara-amanda) shared the following with Juan on 7/19/2023**

> I checked and Angela Fowler is going to be testing this and will be providing further guidance.
>
> **In the meantime here are some resource links relating to legends and labels:**
>     - [Radio button grouping a11y considerations](https://design.va.gov/components/form/radio-button#accessibility-considerations) (DST site)
    - [Making labels and legends headings](https://design-system.service.gov.uk/get-started/labels-legends-headings/) (referencing Platform's comment re: VA Forms team in ticket 59484.

### Type of issue

- [X] Screenreader
- [ ] Keyboard
- [ ] Focus
- [X] Headings
- [ ] Color
- [ ] Zoom
- [ ] Semantics
- [ ] Axe-core
- [ ] Something else
